backend/src/routes/userAccount.ts

```diff
@@ -0,0 +1,310 @@
+import { Connection, Repository } from 'typeorm'
+import { Request, RequestHandler, Response } from 'express'
+
+import { UserAccount } from '../entity/UserAccount'
+import { Permission, PermissionType, permissionTypeFromString } from '../entity/Permission'
+import { Site } from '../entity/Site'
+
+interface UserAccountInterface {
+  userId?: number
+  username?: string
+}
+
+interface PermissionInterface {
+  permission?: string
+  siteId?: string | null
+  userAccounts?: UserAccountInterface[]
+}
+
+interface UserPermissionInterface {
+  permission?: string
+  siteId?: string | null
+}
+
+interface UserAccountPermissionInterface {
+  id?: number
+  username?: string
+  permissions?: UserPermissionInterface[]
+}
+
+export class UserAccountRoutes {
+  private userAccountRepository: Repository<UserAccount>
+  private permissionRepository: Repository<Permission>
+  private siteRepository: Repository<Site>
+
+  constructor(conn: Connection) {
+    this.userAccountRepository = conn.getRepository<UserAccount>('user_account')
+    this.permissionRepository = conn.getRepository<Permission>('permission')
+    this.siteRepository = conn.getRepository<Site>('site')
+  }
+
+  postUser: RequestHandler = async (req: Request, res: Response) => {
+    // Expects valid array of "user:passwordHash" strings in the req.body
+    // Validity should be checked in middleware earlier
+    for (let credentialString of req.body) {
+      let username: string
+      let passwordHash: string
+      ;[username, passwordHash] = credentialString.split(':')
+      try {
+        await this.userAccountRepository.save({
+          username: username,
+          passwordHash: passwordHash,
+        })
+      } catch (err) {
+        console.log(err)
+        res.status(400).send('Bad request: cannot save user into the database\n')
+        return
+      }
+    }
+
+    res.status(200).send('Users created successfully\n')
```

Comment:
Yleensä POST palauttaisi lisätyt käyttäjät samassa muodossa kuin GET. Näin rajapinnan käyttäjä saa uusien käyttäjien id:t.